### PR TITLE
Clarify Setting Message Properties on Preconnections and Connections

### LINKED ISSUE #808: Be more clear about Setting Message Properties on Preconnections and Connections
In 3.2 it is clear that you can set Message Properties on Connections and Preconnections, but the rest of the document is not very clear about this.

In 4.2 for instance the text only talks about setting Selection Properties and Connection Properties on PreConnections. 

In 7.1.3 the text says "Connection Properties describe the default behavior for all Messages on a Connection". But I think that Message Properties set on a Connection or PreConnection may also describe the default behavior so this relationship should be clarified. Also, the Message Property defaults given for Properties that do not depend on other Properties sets the defaults for all Messages in a Connection I assume? I also assume that a Message Property set on a Message Context overrides a Message Property set on a PreConnection or Connection, but this is also not specified.

In 5.4. Connection Groups, I assume that setting Message Properties on a Connection applies to all Connections in the Connection Group? This is not clear.

### PR DESCRIPTION
**Pull request by philsbln**
closes #808

### CODE REVIEW

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -821,8 +821,8 @@ RemoteSpecifier.WithIPv4Address(192.0.2.22)
 A Preconnection Object holds properties reflecting the application's
 requirements and preferences for the transport. These include Selection
 Properties for selecting protocol stacks and paths, as well as Connection
-Properties for configuration of the detailed operation of the selected Protocol
-Stacks.
+Properties and Message Properties for configuration of the detailed operation
+of the selected Protocol Stacks on a per connection and message level.
```
Comment (britram) [inline draft-ietf-taps-interface.md:825]: Might be nitpicking here, but -- the Message Properties at this time are not configuring the Protocol Stack, they're just setting common defaults for Send, correct?

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -1624,7 +1624,7 @@ Preconnection -> EstablishmentError<reason?>
 
 ## Connection Groups {#groups}
 
-Connection Groups can be created using the Clone Action:
+Connection Groups represent sets of entagled connections, e.g., streams in a multi-streaming protocol, and can be created using the Clone Action:
```
Comment (britram) [inline draft-ietf-taps-interface.md:1627]: This change seems unrelated to the main change (and we've gone back and forth here); propose removing it.

**Review by britram — CHANGES_REQUESTED** [review]

File: draft-ietf-taps-interface.md (hunk 1 above)
Comment (philsbln) [reply draft-ietf-taps-interface.md:825]: Correct, but there is no implication when the configuration will actually take effect.

File: draft-ietf-taps-interface.md (hunk 2 above)
Comment (philsbln) [reply draft-ietf-taps-interface.md:1627]: Reading it without this hint felt a little confusing to me after not reading it a long time. I am happy to remove it, but would prefer third opinion here.

File: draft-ietf-taps-interface.md (hunk 1 above)
Comment (mwelzl) [reply draft-ietf-taps-interface.md:825]:
```suggestion
of the selected Protocol Stacks on a per-Connection and Message level.
```

I agree that @britram is nitpicking  :-)   ... because I think that pointing out that Message Properties only influence the send call will only make this harder to read, at no real benefit.

File: draft-ietf-taps-interface.md (hunk 2 above)
Comment (mwelzl) [reply draft-ietf-taps-interface.md:1627]: I agree to remove it. The text below the Clone call explains it all, and it was revised not too long ago in a PR, so this sits with agreement behind it.

A side note, we no longer talk about Connections being "entangled": the term only applies to their Connection Properties.  Also, using the term here before it's explained isn't good... at best, we could use the "streams in a multi-streaming protocol" example, but if you're missing this, it also fits better as an addition to the text below the call.

File: draft-ietf-taps-interface.md [hunk 3]
```diff
@@ -1647,7 +1647,9 @@ to the new Connection when calling Clone(), but in this case, a later change to
 `Connection Priority` on one Connection does not change it on the
 other Connections in the same Connection Group.
 
-Message Properties are also not entangled.  For example,
+Message Properties set on a Connection are also not entangled. Chaning these on an individual Connection
+which is part of a Connections Group SHOULD have the same effect as setting the Message property on each
+Message sent though that Connection. For example,
```
Comment (mwelzl) [inline draft-ietf-taps-interface.md:1652]: The sentence that you included here is only making things more confusing, IMO. Suggest to remove it.

**Review by mwelzl — CHANGES_REQUESTED** [review]
I also wonder if this really does all that's requested in #808...

File: draft-ietf-taps-interface.md (hunk 2 above)
Comment (abrunstrom) [reply draft-ietf-taps-interface.md:1627]: Agree to remove.

File: draft-ietf-taps-interface.md (hunk 3 above)
Comment (abrunstrom) [reply draft-ietf-taps-interface.md:1652]: I think a first modified sentence is sufficient "Message Properties set on a Connection also apply only to that Connection". Actually reading it again now I realize that it is the example that confuses things, because that talks about setting a Message Property on a Message, so maybe we can just remove the example as it does not belong in this section.

File: draft-ietf-taps-interface.md (hunk 2 above)
Comment (philsbln) [reply draft-ietf-taps-interface.md]: Removed it

File: draft-ietf-taps-interface.md (hunk 3 above)
Comment (philsbln) [reply draft-ietf-taps-interface.md]: applied

**Comment by philsbln** [comment]
Applied requested changes

**Review by mwelzl — APPROVED** [review]
Thanks!!

**Review by britram — APPROVED** [review]
